File: php/testcode/unsafereflect_048.php
<?php
require_once __DIR__ . '/shared.php';

class DbConnection {}

// vuln-code-snippet start php_reflect_di_typesafe_resolution
function unsafereflect048(BenchmarkRequest $req): BenchmarkResponse {
    $container = new stdClass();
    $container->bind = function(string $iface, string $concrete) {};
    $svc = new UserService(new DbConnection()); // vuln-code-snippet safe-line php_reflect_di_typesafe_resolution
    return BenchmarkResponse::ok('resolved');
}
// vuln-code-snippet end php_reflect_di_typesafe_resolution

File: ruby/testcode/unsafereflect_048.rb
require_relative 'shared'

COMMANDS = { 'create' => -> { 'created' }, 'delete' => -> { 'deleted' } }.freeze

# vuln-code-snippet start ruby_reflect_command_bus
def handler(req)
  result = COMMANDS.fetch(req.param('cmd')).call # vuln-code-snippet safe-line ruby_reflect_command_bus
  BenchmarkResponse.json({ result: result })
end
# vuln-code-snippet end ruby_reflect_command_bus
